### PR TITLE
Feature/generic payloads

### PR DESCRIPTION
**Pull request by TalhaAli00**

### CODE REVIEW

File: sample/src/main/java/org/walletconnect/example/payloads/EthSendTransaction.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.walletconnect.example.payloads
+
+data class EthSendTransaction(
```
Comment (chaak) [inline sample/src/main/java/org/walletconnect/example/payloads/EthSendTransaction.kt:3]: If not used should be deleted

File: sample/src/main/java/org/walletconnect/example/payloads/EthSign.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package org.walletconnect.example.payloads
+
+data class EthSign(
```
Comment (chaak) [inline sample/src/main/java/org/walletconnect/example/payloads/EthSign.kt:3]: Same

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/client/WalletConnectClientData.kt [hunk 3]
```diff
@@ -20,10 +23,30 @@ sealed class WalletConnectClientData {
 
     data class SessionRequest(
         val topic: String,
-        val request: Any,
+        val requestStringified: String,
         val chainId: String?,
         val method: String
-    )
+    ) {
+//        @ExperimentalStdlibApi
```
Comment (chaak) [inline walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/client/WalletConnectClientData.kt:30]: Again, we shouldn't keep code which is commented

File: walletconnectv2/src/test/kotlin/org/walletconnect/walletconnectv2/GenericPayloadsTest.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+import com.squareup.moshi.*
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import org.intellij.lang.annotations.Language
+import org.junit.jupiter.api.Test
+import java.lang.reflect.Type
+import java.util.*
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
```
Comment (chaak) [inline walletconnectv2/src/test/kotlin/org/walletconnect/walletconnectv2/GenericPayloadsTest.kt:11]: What about this class? If it's not relied to our production code it should be deleted. It creates only confusion. When we go back to the generic adapter issue we're gonna start from scratch anyways.

**Review by chaak — CHANGES_REQUESTED** [review]
See below my comments

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/client/WalletConnectClientData.kt (hunk 3 above)
Comment (TalhaAli00) [reply walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/client/WalletConnectClientData.kt:30]: agreed

File: walletconnectv2/src/test/kotlin/org/walletconnect/walletconnectv2/GenericPayloadsTest.kt (hunk 4 above)
Comment (TalhaAli00) [reply walletconnectv2/src/test/kotlin/org/walletconnect/walletconnectv2/GenericPayloadsTest.kt:11]: so I don't want to lose the train of thought with the work that was started. Once we come back to it, we can remove it if there's another approach we want to take

**Review by chaak — APPROVED** [review]

**Pull request merged**